javalib/src/main/scala/java/util/ArrayList.scala

```diff
@@ -0,0 +1,84 @@
+package java.util
+
+import scala.scalajs._
+
+class ArrayList[E] private (
+    inner: js.Array[Box[E]]
+  ) extends AbstractList[E]
+    with RandomAccess
+    with Cloneable
+    with Serializable { self =>
+
+  def this() {
+    this(new js.Array[Box[E]])
+  }
+
+  def this(initialCapacity: Int) {
+    this(new js.Array[Box[E]])
+  }
+
+  def this(c: Collection[_ <: E]) {
+    this(new js.Array[Box[E]])
+    addAll(c)
+  }
+
+  override def clear(): Unit =
+    inner.clear()
+
+  override def add(e: E): Boolean = {
+    inner.push(Box(e))
+    true
+  }
+
+  override def add(index: Int, element: E): Unit = {
+    checkIndexOnBounds(index)
+    inner.splice(index, 0, Box(element))
+  }
+
+  def get(index: Int): E = {
+    checkIndexInBounds(index)
+    inner(index).inner
+  }
+
+  def size(): Int =
+    inner.length
+
+  override def set(index: Int, element: E): E = {
+    val e = get(index)
+    inner.update(index, Box(element))
+    e
+  }
+
+  override def remove(index: Int): E = {
+    val e = get(index)
+    inner.splice(index, 1)
+    e
+  }
+
+  override def subList(fromIndex: Int, toIndex: Int): List[E] =
+    new ArrayList(inner.jsSlice(fromIndex, toIndex))
```

Comment:
as per:
https://docs.oracle.com/javase/7/docs/api/java/util/ArrayList.html#subList(int,%20int)

subList must be backed on the original Array, creating a new one will work (I'm facing the same problem in another list and solving it in this way:
https://github.com/andreaTP/scala-js/blob/master/javalib/src/main/scala/java/util/concurrent/ConcurrentSkipListSet.scala#L86
but if you have a best solution please tell!
)